[IMP] account_payment_partner: Fill payment mode in invoices if none is provided

pedrobaeza opened the pull request:
Using same method as in upstream, payment mode is filled on invoice creation if no payment method is provided. This way, we don't need to install _account_payment_sale_ if we don't want to handle several payment modes at sales level. Even more, if we install the module later and we have already existing
sales orders without payment mode filled, those orders will be invoiced with the customer payment mode.

pedrobaeza commented:
@sbidoul @alexis-via please review

cc @Tecnativa

sbidoul commented:
I don't understand the goal of this PR. This looks like this could have surprising side effects. What if, for instance, one wants to create an invoice without payment mode?

pedrobaeza commented:
The goal is explained in the first comment. Please read it again.

What you say is very simple to achieve: just include 'payment_mode_id': False in your vals. As stated, Odoo makes the same with acount_id, payment_term_id, and so on. There's no possibility of side effects, as you only recover the specific field (payment_mode_id). Not others.

pedrobaeza commented:
I have added the same logic for filling automatically the mandate.

@sbidoul do you have clear my purpose? Any other doubt?

luismontalba reviewed (approved):
Tested on runbot

pedrobaeza commented:
@sbidoul @alexis-via can you please check again?

sbidoul commented:
I see that it is indeed done this way [upstream](https://github.com/odoo/odoo/blob/963c9ffa4ae554df0ea0220e31a3b67dd4a8d0bb/addons/account/models/account_invoice.py#L329).

What a ugly hack... I hope it will not propagate everywhere there is a create method.

Since this changes the behaviour, can you bump the version number?

pedrobaeza committed:
[IMP] account_payment_partner: Fill payment mode in invoices if none is provided

Using same method as in upstream, payment mode is filled on invoice creation if
no payment method is provided. This way, we don't need to install
account_payment_sale if we don't want to handle several payment modes at sales
level. Even more, if we install the module later and we have already existing
sales orders without payment mode filled, those orders will be invoiced with
the customer payment mode.

pedrobaeza committed:
[IMP] account_banking_mandate: Fill mandate in invoices if none is provided

Using same method as in upstream, mandate is filled on invoice creation if no
one is provided. This way, we don't need to install account_banking_mandate_sale
if we don't want to handle several mandates at sales level.

pedrobaeza commented:
Done, thanks.

sbidoul commented:
@pedrobaeza I meant the minor version (this is an api change, as the behaviour of create() changes).

pedrobaeza commented:
Well, it's not an API change (arguments continue the same), nor it requires a module update. That's why I only changed the revision number.

May you proceed with the merge?

sbidoul commented:
@pedrobaeza if the behaviour of a method changes (other than bugfixes of course), even if the signature remains the same, then it's an API change.

chienandalu reviewed (approved):
tested on runbot

pedrobaeza commented:
@sbidoul version number changed to minor. Please merge.

pedrobaeza commented:
Thanks!

pedrobaeza commented:
Cherry-picking on v10
